NebulaWorld/SimulatedWorld.cs

```diff
@@ -227,10 +227,17 @@ public static void OnGameLoadCompleted()
             // Assign our own color
             UpdatePlayerColor(LocalPlayer.PlayerId, LocalPlayer.Data.Color);
 
-            // TODO: Investigate where are the weird position coming from ?
-            // GameMain.mainPlayer.transform.position = data.Position.ToUnity();
-            // GameMain.mainPlayer.transform.eulerAngles = data.Rotation.ToUnity();
-
+            // Change player location from spawn to the last known
+            VectorLF3 UPosition = new VectorLF3(LocalPlayer.Data.UPosition.x, LocalPlayer.Data.UPosition.y, LocalPlayer.Data.UPosition.z);
+            if (UPosition != VectorLF3.zero)
+            {
+                GameMain.mainPlayer.planetId = LocalPlayer.Data.LocalPlanetId;
+                GameMain.mainPlayer.position = LocalPlayer.Data.LocalPlanetPosition.ToUnity(); ;
+                GameMain.mainPlayer.uPosition = UPosition;
+                GameMain.mainPlayer.uRotation = Quaternion.Euler(LocalPlayer.Data.Rotation.ToUnity());
+                GameMain.mainPlayer.transform.localScale = Vector3.one;
```

Comment:
did you test this when spawning on a remote planet? Afaik you ened to set the `localScale` after the player was set as child to the planet (iirc this is what the game does when you are on a planet) which usually is after `ArrivePlanet()`, else you could end up with weird icarus sizes.

Reply:
You are maybe right, I am checking it now. There is also some issue that if I leave the planet and come back I will be always spawned in the air. Hmm...

Reply:
maybe you need to recalculate the position based on planet movement, something like [this](https://github.com/sp00ktober/DSP_Mod/blob/2becb2f4abf7d280db6c935d1c65e499c6bed676/RenderDistance/Class1.cs#L230)

Reply:
Go to 0:25s [video](https://www.youtube.com/watch?v=Lzpml9i8a24).
Seems like the player's transform parent is not the planet. Still looking into it.

Reply:
i think this is because you are not setting `localPlanet` (look at the globe in the corner it tells the wrong planet).
Thats why you start floating as the game does not think you are on that planet right now, it does not even load the `FactoryData` if im seeing this right.

Reply:
right!

Reply:
adding ` GameMain.data.ArrivePlanet(GameMain.data.galaxy.PlanetById(LocalPlayer.Data.LocalPlanetId));` seems to solve it. Test cases in [video](https://youtu.be/XlQIEUJvmNw):

- Load player data on the same planet
- Load player data from the space
- Load player data from another planet

Reply:
oh wait, I will have to change it to `ArriveStar()` :(